### DISCUSSION TITLE
Is there support for registering custom methods that can be called in function node or other nodes?

### DISCUSSION THREAD

**Discussion by chengguotao** [gorules/zen#199] [Q&A] [answered]
Dear development team：
       Do I currently or in the future support custom method calls through registration, like other rule engines or scripting languages, because I want to use this way, to implement complex business logic, such as using transactions to call databases or other services。
        Looking forward to your reply！！

**Comment by ivanmiletic** [comment]
Hi @chengguotao ,

I suppose you are talking about BRE, not BRMS.

You are able to register custom methods, it is not yet documented but you can create a customNode that can call your custom logic as a callback method. It is available in all languages.

For Editor you can see an example here: https://github.com/gorules/jdm-editor/blob/master/packages/jdm-editor/src/components/decision-graph/dg.stories.tsx
Engine Example (NodeJS):
```
const engine = new ZenEngine({ customHandler: async (request: ZenEngineHandlerRequest) => {
  return {};
} })
```

Best regards,
Ivan

**Comment by chengguotao** [comment, marked as answer]
I'm trying to register rust's method in BRE and call it in the funcation declaration in BRMS

**Reply by chengguotao** [reply]
I know the answer：https://github.com/gorules/zen/discussions/124#discussioncomment-8976890
